mutable/slice.go

```diff
@@ -0,0 +1,95 @@
+package mutable
+
+import "math/rand"
+
+// Filter iterates over elements of collection, returning an array of all elements predicate returns truthy for.
+// The function returns the modified slice, which may be shorter than the original if some elements were removed.
+// The order of elements in the original slice is preserved in the output.
+// Play:
+func Filter[T any](collection *[]T, predicate func(item T) bool) {
+	FilterI(collection, func(item T, index int) bool {
+		return predicate(item)
+	})
+}
+
+// Filter iterates over elements of collection, returning an array of all elements predicate returns truthy for.
+// The function returns the modified slice, which may be shorter than the original if some elements were removed.
+// The order of elements in the original slice is preserved in the output.
+// Play:
+func FilterI[T any](collection *[]T, predicate func(item T, index int) bool) {
```

Comment:
Compared to immutable package, I chose to split `Filter` into 2 helpers: the `I` means predicate receives and index as parameter.

Reply:
I would not use the I name and use your current FilterI as Filter

There is an index, I don't use it, let's user ignore it. You have no need to provide a variant of all your methods